atlasdb-config/src/main/java/com/palantir/atlasdb/http/VersionSelectingClients.java

```diff
@@ -83,6 +84,19 @@ private VersionSelectingClients() {
                 errorMetric::increment);
     }
 
+    static <T> T createRefreshingNewClient(
+            Supplier<TargetFactory.InstanceAndVersion<T>> newClientSupplier,
+            Class<T> clazz) {
+
+        return ExperimentRunningProxy.newProxyInstance(
+                Suppliers.compose(TargetFactory.InstanceAndVersion::instance, newClientSupplier::get),
+                null,
```

Comment:
Is ok, since `useExperimental` is true and `useFallback` false. Is not great, but maybe exploding if we get into this kind of state that we don't expect is possible is not the worst reaction.